### PR TITLE
feat: add new program certificate events

### PR DESCRIPTION
**Pull request by justinhynes**
**Description:** 

Internal 2U ticket: APER-2625

This PR adds new `OpenEdxPublicSignal` events triggered by program certificate generation and revocation.

- Adds a `PROGRAM_CERTIFICATE_AWARDED` event
- Adds a `PROGRAM_CERTIFICATE_REVOKED` event
- Adds new `ProgramCertificateData` and `ProgramData` data classes to support the new events

**ISSUE:** https://github.com/openedx/openedx-events/issues/250

**Reviewers:**

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** None at this time.

### CODE REVIEW

File: openedx_events/management/commands/generate_avro_schemas.py
Comment (justinhynes) [inline openedx_events/management/commands/generate_avro_schemas.py:32]: Updated docs here to remove `cms` as this will cause an error when running the management command.

Comment (robrap) [reply openedx_events/management/commands/generate_avro_schemas.py:32]: @rgraber: Just checking whether this was just a typo in the orginal docs, or whether this had significance that we need to look into. Thanks.

Comment (rgraber) [reply openedx_events/management/commands/generate_avro_schemas.py:32]: Hmm. This is how you run the command in edx-platform but not anything else, because edx-platform requires you to specify lms or cms when running any management command. Maybe we should have 2 examples, one for running in edx-platform and one for running somewhere else?

File: openedx_events/learning/data.py
Comment (hurtstotouchfire) [inline openedx_events/learning/data.py:287]: learner
```suggestion
        certificate_available_date (datetime): Optional. A DateTime describing when a learner is allowed to view the
```

File: openedx_events/learning/data.py
Comment (hurtstotouchfire) [inline openedx_events/learning/data.py:289]: iirc we define this set of supported statuses somewhere in Credentials. Do we have any way of referencing that in comments like this?

File: openedx_events/learning/data.py
Comment (hurtstotouchfire) [inline openedx_events/learning/data.py:270]: Same comment as [L289](https://github.com/openedx/openedx-events/pull/253/files#r1276540953). Possible values are defined... I wanna say in course-discovery?? 😭

File: openedx_events/event_bus/avro/tests/schemas/org+openedx+learning+program+certificate+revoked+v1_schema.avsc
Comment (hurtstotouchfire) [inline openedx_events/event_bus/avro/tests/schemas/org+openedx+learning+program+certificate+revoked+v1_schema.avsc:27]: whoa neat is `pii` a data type in this schema?

File: openedx_events/event_bus/avro/tests/schemas/org+openedx+learning+program+certificate+awarded+v1_schema.avsc
Comment (hurtstotouchfire) [inline openedx_events/event_bus/avro/tests/schemas/org+openedx+learning+program+certificate+awarded+v1_schema.avsc:89]: what does this default do?

File: openedx_events/learning/data.py
Comment (justinhynes) [reply openedx_events/learning/data.py:287]: Nice catch! Thank you.

File: openedx_events/learning/data.py
Comment (justinhynes) [reply openedx_events/learning/data.py:289]: In Credentials we only have two statuses -- `awarded` or `revoked`. Course Certificates in the LMS support many more statuses. 

https://github.com/openedx/credentials/blob/7a08a267605a3d1e913ec81a3566609ffd09ac8f/credentials/apps/credentials/models.py#L158-L161.

I can add a link to this if you think it's useful!

File: openedx_events/learning/data.py
Comment (justinhynes) [reply openedx_events/learning/data.py:270]: They are and they are synced to Credentials as part of the `copy_catalog` workflow. We have access to them all in Credentials. I'm not sure where they are defined in Discovery but I can add a link to them if I find it if you'd like!

File: openedx_events/management/commands/generate_avro_schemas.py
Comment (robrap) [reply openedx_events/management/commands/generate_avro_schemas.py:32]: @rgraber: Actually, is there a reason why this needs to be a management command run from elsewhere, rather than just a script that could be run from openedx-events? I don't get why we'd want to involve any other repo in the creation of tests from this repo, but maybe that is required for some reason.

File: openedx_events/event_bus/avro/tests/schemas/org+openedx+learning+program+certificate+revoked+v1_schema.avsc
Comment (justinhynes) [reply openedx_events/event_bus/avro/tests/schemas/org+openedx+learning+program+certificate+revoked+v1_schema.avsc:27]: The `UserData` data class is made up of non-PII and PII data. Relevant links: 

UserData: https://github.com/openedx/openedx-events/blob/c0eb4ba1a3d7d066d58e5c87920b8ccb0645f769/openedx_events/learning/data.py#L45
UserNonPersonalData: https://github.com/openedx/openedx-events/blob/c0eb4ba1a3d7d066d58e5c87920b8ccb0645f769/openedx_events/learning/data.py#L15
UserPersonalData: https://github.com/openedx/openedx-events/blob/c0eb4ba1a3d7d066d58e5c87920b8ccb0645f769/openedx_events/learning/data.py#L29

**Comment by robrap** [comment]
@justinhynes 
1. [tip] Add `fixup! ` or `squash! ` to your commit title to get a passing `Lint Commit Messages` check before squashing. Or, if you like it to be red to remind you to squash, ignore my tip. 😉 
2. I added `waiting for eng review` label for arch-bom review. I'm trying this process out. Let me know if this ends up being a process problem. Thanks.

**Comment by felipemontoya** [comment]
@mariajgrimaldi and I are requested review automatically for being the only members of the [maintainers group](https://github.com/orgs/openedx/teams/hooks-extension-framework). We try to be quick to address this reviews, but sometimes we aren't.
I believe however that anyone that is a [core contributor](
https://openedx.atlassian.net/wiki/spaces/COMM/pages/3156344833/Core+Contributors+to+the+Open+edX+Project) to this repo would qualify to give reviews. If they are unsure we can always be tagged specifically.

This particular PR I just reviewed and It looks good and all outstanding comments have been addressed so I'll leave the approval for when the Linting check passes.

**Review by felipemontoya — APPROVED** [review]
The definitions and naming look good to me. The Avro part I can't verify but it looks reasonable.

File: openedx_events/event_bus/avro/tests/schemas/org+openedx+learning+program+certificate+awarded+v1_schema.avsc
Comment (justinhynes) [reply openedx_events/event_bus/avro/tests/schemas/org+openedx+learning+program+certificate+awarded+v1_schema.avsc:89]: I realized I read this comment wrong. My brain interpreted it as "what does this default TO" which is definitely not what it says. 

Actually answering your question, I'll take the explanation from the Avro spec doc:
```
default: A default value for this field, used when reading instances that lack this field (optional). Permitted values depend on the field's schema type, according to the table below. Default values for union fields correspond to the first schema in the union. Default values for bytes and fixed fields are JSON strings, where Unicode code points 0-255 are mapped to unsigned 8-bit byte values 0-255.
```

File: openedx_events/management/commands/generate_avro_schemas.py
Comment (rgraber) [reply openedx_events/management/commands/generate_avro_schemas.py:32]: It shouldn't require any other repo. We could move it to a standalone script.

Comment (robrap) [reply openedx_events/management/commands/generate_avro_schemas.py:32]: If it runs here, then all is well. I just didn’t understand the `cms`  comment. All good.

**Pull request merged**
